src/name/name.rs

```diff
@@ -45,17 +45,11 @@ impl<'a> SubjectNameRef<'a> {
     /// consider parsing these with `std::net::IpAddr` and then using
     /// `IpAddr::from<std::net::IpAddr>`.
     pub fn try_from_ascii(subject_name: &'a [u8]) -> Result<Self, InvalidSubjectNameError> {
-        if ip_address::is_valid_ipv4_address(untrusted::Input::from(subject_name)) {
-            return Ok(SubjectNameRef::IpAddress(IpAddrRef::V4(
-                subject_name,
-                ip_address::ipv4_octets(subject_name).map_err(|_| InvalidSubjectNameError)?,
-            )));
+        if let Ok(ip_address) = ip_address::parse_ipv4_address(subject_name) {
+            return Ok(SubjectNameRef::IpAddress(ip_address));
         }
-        if ip_address::is_valid_ipv6_address(untrusted::Input::from(subject_name)) {
-            return Ok(SubjectNameRef::IpAddress(IpAddrRef::V6(
-                subject_name,
-                ip_address::ipv6_octets(subject_name).map_err(|_| InvalidSubjectNameError)?,
-            )));
+        if let Ok(ip_address) = ip_address::parse_ipv6_address(subject_name) {
```

Comment:
Can do `else if` here.

Reply:
It's a return-on-first-match situation, no need for it, but I will add it nevertheless.

Reply:
Addressed in 416976fc8ba67324491c9be789c17e48f6fb6405